src/pages/MangaLib/main.ts

```diff
@@ -0,0 +1,300 @@
+/* eslint-disable new-cap */
+/* eslint-disable global-require */
+import { pageInterface } from '../pageInterface';
+import { SyncPage } from '../syncPage';
+import { Manga, getMangaData, getChaptersData, isPageAPI, Chapters } from '../AnimeLib/api';
+import { text } from '../../utils/mangaProgress/modes/text';
+
+const { asyncWaitUntilTrue: awaitReaderLoading, reset: resetAwaitReader } =
+  utils.getAsyncWaitUntilTrue(() => j.$('main img').length);
+
+const { asyncWaitUntilTrue: awaitOverviewLoading, reset: resetAwaitOverview } =
+  utils.getAsyncWaitUntilTrue(() => j.$('.tabs-item').length);
+
+const manga: Manga = {
+  data: {
+    id: 0,
+    name: '',
+    rus_name: '',
+    eng_name: '',
+    slug_url: '',
+    cover: {
+      thumbnail: undefined,
+      default: undefined,
+    },
+  },
+  reader: {
+    chapter: 0,
+    total: 1,
+    volume: 0,
+    total_subchapters: 1,
+    current_subchapter: 0,
+    current_subchapter_index: 0,
+    next: undefined,
+  },
+};
+export const MangaLib: pageInterface = {
+  name: 'MangaLib',
+  // NOTE - Add mangalib.me when domain is updated to new version
+  domain: ['https://mangalib.org', 'https://v2.slashlib.me', 'https://v1.yaoilib.net'],
+  languages: ['Russian'],
+  type: 'manga',
+  getImage() {
+    return manga.data.cover.default || manga.data.cover.thumbnail;
+  },
+  isSyncPage(url) {
+    return utils.urlPart(url, 5) === 'read' && !isPageAPI(url);
+  },
+  isOverviewPage(url) {
+    return utils.urlPart(url, 4) === 'manga' && !isPageAPI(url);
+  },
+  sync: {
+    getTitle(url) {
+      return manga.data.eng_name || manga.data.name || manga.data.rus_name;
+    },
+    getIdentifier(url) {
+      return manga.data.id.toString();
+    },
+    getOverviewUrl(url) {
+      return utils.absoluteLink(`ru/manga/${manga.data.slug_url}`, MangaLib.domain);
+    },
+    getEpisode(url) {
+      return manga.reader.chapter;
+    },
+    getVolume(url) {
+      return manga.reader.volume || 1;
+    },
+    nextEpUrl(url) {
+      return manga.reader.next;
+    },
+    readerConfig: [
```

Comment:
I feel like this is a bit overcomplicated. I think it would be better to just disable updates generally if not at the last subchapter and use the normal page based config on the last subchapter. What do you think about that?


Currently the condition is only a dom selector. But it could easily be extended to an boolean function. 

```
readerConfig: [
      {
        condition: () => {
          if (not float) return false;
          if (not able to get total_subchapters or 0 or whatever is important) return false;
          return manga.reader.total_subchapters > manga.reader.current_subchapter_index! + 1;
        },
        {
        current: {
          mode: 'callback',
          callback: () => 0,
        },
        total: {
          mode: 'callback',
          callback: () => 2,
        },
      },
      {
        current: {
          selector: 'footer',
          mode: 'text',
          regex: '(\\d+) / (\\d+)$',
          group: 1,
        },
        total: {
          selector: 'footer',
          mode: 'text',
          regex: '(\\d+) / (\\d+)$',
          group: 2,
        },
      },
    ],
```

Reply:
I guess what you are talking is more simpler, right. I'll try it and check how it feels, I might change it to what you mean in both PRs since I did it the same way

Reply:
Btw how did you write condition as function? It's only accepts `string`